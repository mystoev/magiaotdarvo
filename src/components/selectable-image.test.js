import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import SelectableImage from './selectable-image';

describe('<SelectableImage /> tests', () => {
  it('should render', () => {
    const { container } = render(<SelectableImage />);

    expect(container).toMatchSnapshot();
  });

  it('should render with image', () => {
    const { container } = render(
      <SelectableImage category={'cat1'} productName={'name1'} file={'file1'} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should call onSelect when clicked', () => {
    const onSelectMock = jest.fn();
    const { queryByRole } = render(
      <SelectableImage
        category={'cat1'}
        productName={'name1'}
        file={'file2'}
        onSelect={onSelectMock}
      />
    );

    fireEvent.click(queryByRole('img'));

    expect(onSelectMock).toHaveBeenCalledWith({ file: 'file2', isSelected: true });
  });
});
