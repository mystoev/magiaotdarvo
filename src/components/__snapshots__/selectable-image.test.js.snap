// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectableImage /> tests should render 1`] = `
<div>
  <div
    class="selectable-image__SelectableImageContainer-sc-1px0fdw-0 kQTPRI"
  >
    <div
      class="selectable-image-overlay normal"
    />
    <img
      class="selectable-image"
      src=""
    />
  </div>
</div>
`;

exports[`<SelectableImage /> tests should render with image 1`] = `
<div>
  <div
    class="selectable-image__SelectableImageContainer-sc-1px0fdw-0 kQTPRI"
  >
    <div
      class="selectable-image-overlay normal"
    />
    <img
      class="selectable-image"
      src="http://localhost:8080/images/cat1/name1/file1"
    />
  </div>
</div>
`;
